mathics/core/atoms.py

```diff
@@ -239,11 +239,14 @@ def __new__(cls, value) -> "Integer":
         return self
 
     def __eq__(self, other) -> bool:
-        return (
-            self._value == other.value
-            if isinstance(other, Integer)
-            else super().__eq__(other)
-        )
+        if isinstance(other, Integer):
+            return self._value == other.value
+        if isinstance(other, Number):
```

Comment:
Here, numbers are not treated as generic expressions in checking equality: This fixes the issue with `RGBColor[1,0,0]==RGBColor[1.,0,0]` keeping the right order in ```Sort[{1.,1,1.`3, 1.`9}]```